test/sh/failpods.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+#
+# Copyright (c) 2022. Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+#
+
+namespace=""
+maxPods=10
+
+for param in $*; do
+  case $param in
+  "--ns")
+    shift
+    namespace=$1
+    shift
+    ;;
+  "--timeoutseconds")
+    shift
+    timeoutseconds=$1
+    shift
+    ;;
+  "--help")
+    shift
+    echo "parameters: --ns driver_namespace [--timeoutseconds value]"
+    exit
+    ;;
+  esac
+done
+
+[ "$namespace" = "" ] && echo "Required argument: --ns driver_namespace --timeoutseconds value" && exit 2
+
+# nodeList returns a list of nodes
+nodeList() {
+	kubectl get nodes -A | grep -v 'mast.r'  | grep -v NAME | awk '{ print $1 }'
+}
+
+# getNumOfPods returns the number of pods on a node $1 for a specific namespace
+getNumOfPods() {
+	# Add an extra space on match string to differentiate worker-1 from worker-10
+	kubectl get pods -A -o wide | grep "$1 " | grep $namespace | wc -l 
+}
+
+# getWorker returns the first worker node that we are targeting
+getWorker(){
+    kubectl get nodes -A | grep -v 'mast.r'  | grep -v NAME | awk 'NR==1{ print $1 }'
```

Comment:
Hi Shayna, it looks like you're using this line to filter out the master nodes. This will work fine on k8s 1.23 and below, but may not work on 1.24 and above, as kubernetes is going through their own process of removing objectionable terms. Here is what I see on my k8s 1.24 system for the nodes:
[root@master tom]# k get nodes -o wide
NAME              STATUS   ROLES           AGE     VERSION   INTERNAL-IP      EXTERNAL-IP   OS-IMAGE                               KERNEL-VERSION          CONTAINER-RUNTIME
master.domain     Ready    **control-plane**   7d17h   v1.24.0   10.247.102.211   <none>        Red Hat Enterprise Linux 8.4 (Ootpa)   4.18.0-305.el8.x86_64   containerd://1.6.4
worker-1.domain   Ready    <none>          7d17h   v1.24.0   10.247.102.213   <none>        Red Hat Enterprise Linux 8.4 (Ootpa)   4.18.0-305.el8.x86_64   containerd://1.6.4
worker-2.domain   Ready    <none>          7d17h   v1.24.0   10.247.102.215   <none>        Red Hat Enterprise Linux 8.4 (Ootpa)   4.18.0-305.el8.x86_64   containerd://1.6.4

The filtering would work on my machine because the node name is master.domain, but notice the ROLES column no longer has master, but has control-plane instead. You might want the grep to be grep -v -E 'mast.r|control-plane'.
However I'll leave this as informative and you can decide.
Please feel free to check the script on my system 10.247.102.211 root / dangerous.
Tom

Reply:
Thank you for catching that. Fixed.